language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/ConfigTomlValueGenerationUtil.java

```diff
@@ -0,0 +1,163 @@
+     * @param basicType union of basic types
+     * @param anydataOrJson union of anydata and json
+     * @param confName name of the configurable
+     * @return value string wrapped in {@link  TomlEntryValue}
+     */
+    public static TomlEntryValue getDefaultValueStr(TypeSymbol type, TypeSymbol basicType,
+                                            TypeSymbol anydataOrJson, String confName) {
+        switch (type.typeKind()) {
+            case BOOLEAN -> {
+                return new TomlEntryValue("false", true);
+            }
+            case BYTE, INT, INT_SIGNED8, INT_SIGNED32, INT_UNSIGNED8, INT_UNSIGNED16, INT_UNSIGNED32, INT_SIGNED16 -> {
+                return new TomlEntryValue("0", true);
+            }
+            case FLOAT, DECIMAL -> {
+                return new TomlEntryValue("0.0", true);
+            }
+            case STRING, XML, ANYDATA, JSON, SINGLETON -> {
+                return new TomlEntryValue("\"\"", true);
+            }
+            case TUPLE -> {
+                return new TomlEntryValue("[]", true);
+            }
+            case ARRAY -> {
+                ArrayTypeSymbol arrayTypeSymbol = (ArrayTypeSymbol) type;
+                TypeSymbol memberType = CommonUtil.getRawType(arrayTypeSymbol.memberTypeDescriptor());
+                if (memberType.typeKind() == TypeDescKind.UNION) {
+                    memberType = ((UnionTypeSymbol) memberType).memberTypeDescriptors().get(0);
+                }
+                if (memberType.subtypeOf(basicType) || basicType.subtypeOf(anydataOrJson)) {
+                    return new TomlEntryValue("[]", true);
+                }
+                return new TomlEntryValue(String.format("[[%s]]%n", confName), false); // records and maps
+            }
+            case MAP -> {
+                return new TomlEntryValue(String.format("[%s]%n", confName), false);
+            }
+            case RECORD -> {
+                StringBuilder stringBuilder = new StringBuilder();
+                stringBuilder.append(String.format("[%s]%n", confName));
+                getRecordFields((RecordTypeSymbol) type, stringBuilder, "");
+                return new TomlEntryValue(stringBuilder.toString(), false);
+            }
+            case TABLE -> {
+                return new TomlEntryValue(String.format("[[%s]]%n", confName), false);
+            }
+            case UNION -> {
+                return  getDefaultValueStr(((UnionTypeSymbol) type).memberTypeDescriptors().get(0),
+                        basicType, anydataOrJson, confName);
+            }
+            case TYPE_REFERENCE -> {
+                return getDefaultValueStr(CommonUtil.getRawType(type), basicType, anydataOrJson, confName);
+            }
+            default -> {
+                return new TomlEntryValue("", true);
+            }
+        }
+    }
+
+    private static StringBuilder getRecordFields(RecordTypeSymbol recordTypeSymbol, StringBuilder
+            builder, String prefix) {
```

Comment:
```suggestion
    private static StringBuilder getRecordFields(RecordTypeSymbol recordTypeSymbol, StringBuilder builder,
                                                 String prefix) {
```
Check other places too